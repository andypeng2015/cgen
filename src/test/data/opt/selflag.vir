module foo

export function w $foo(w %x) {
@start:
  %c = eq.w %x, 1_w
  %x = sel.w %c, 1_w, 0_w
  ret %x
}
