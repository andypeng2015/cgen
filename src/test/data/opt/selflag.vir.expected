module foo

export function w $foo(w %x) {
@0:
  %1 = eq.w %x, 0x1_w ; @4
  %0 = flag.w %1 ; @3
  ret %0
}
